### DISCUSSION TITLE
directory/files order in multiline

### DISCUSSION THREAD

**Discussion by Docbroke** [leo-arch/clifm#68] [Ideas]
current
```
1 Books     2 Desktop     3 Documents
4 Downloads     5 Pictures
```
I think it is preferable to have
```
1 Book               4 Downloads
2 Desktop          5 Pictures
3 Documents
```
It is easy to find files this way in long lists as you don't have to move your eyes in zigzag manner.

**Comment by leo-arch** [comment]
I didn't deviate from the ls-like listing mode just because I want it or like it that way, but because of the way files listing is implemented in CliFM. Modifying this is possible, of course, but it requires refactoring one of the most important functions of the program. Besides, this function requires a lot of calculations to correctly display columns. In other words, despite of what it seems at first sight, this one is not an easy task.

However, you're right: listing files in a ls-like manner is just better (or at least some kind of standard). So, this issue is definitely in my TODO list.

**Reply by Docbroke** [reply]
Yes, I understand. This isn't a critical issue either. 
I do like `LongViewMode` with listing in single column, and it is ls-like.

**Reply by leo-arch** [reply]
However, I'm quite close of getting ls-like listing working. Just found an easy way of manipulating ELN's and file names placement. Still a few issues, but close.

If I really manage to make this work, it will be an alternative to the actual listing mode: I'm just used to it.

**Comment by leo-arch** [comment]
The new listing mode is already in master. New option in the config file: `ListingMode`, taking two values: `0` (vertically, `ls(1)`-like), and `1`: horizontally (old mode). Defaults to zero.

Please give it a try as soon as you can: edge cases are there to frustrate even the most amazing and well written code.

**Reply by Docbroke** [reply]
Looks good. I will report if I find anything unusual. Thanks again.

**Comment by Docbroke** [comment]
pager is getting confused with vertical listing.

**Reply by leo-arch** [reply]
Thanks for reporting it. I'll take a look at it.

Now working on a secondary prompt for the internal move command (`m`): something similar to `imv(1)`.

EDIT: The pager seems to be completely screwed for both listing modes.

**Reply by Docbroke** [reply]
> Now working on a secondary prompt for the internal move command (`m`): something similar to `imv(1)`.

This is cool feature. For me `imv` is image viewer program, I wasn't aware of `imv` from `renameutils`. So it took me a while in finding out what changed with `m`

**Reply by leo-arch** [reply]
`imv` was an image viewer for me too, till I tried to open an image with it (after having installed `renameutils`): surprise. What `imv` (as renamer) does is quite simple, but useful. So, why not to implement it for CliFM too? Documentation for the new `m` feature is [here](https://github.com/leo-arch/clifm/wiki/Introduction#c-l-e-edit-m-md-r).

**Comment by leo-arch** [comment]
> pager is getting confused with vertical listing

This issue should be fixed. Working now on a few issues left with horizontal listing mode.
